### ISSUE TITLE
Add bitmap fonts

### ISSUE THREAD

**Issue by 17cupsofcoffee** [17cupsofcoffee/tetra#80]
I previously decided against this in #17, since I thought TTF rendering would suffice, but I'm less sure now - TTFs don't work as well when the screen is scaled from a very small size, which is one of the main ways I'm using Tetra.

**Comment by 17cupsofcoffee** [comment]
The rewrite of the text rendering API in 0.4 is specifically aimed at making this possible - should be able to add it in a patch release once that comes out.

**Comment by 17cupsofcoffee** [comment]
I have started working on this on a branch, as the forced anti-aliasing that `ab-glyph` does isn't suitable for pixel art games like the one I'm making. It works well so far, and the existing layout code handles the metrics from a BMFont fine :)

The main outstanding questions/blockers are:

* There is not a good quality BMFont parser crate that I've found so far - one of them tries to be too high level without exposing the actual data in the file, and the other makes some strange/inefficient API choices (and has some debug logging left in...).
    * I'm quite tempted to write my own, since the format isn't very complex.
* Do we actually need the full `FontCache` for bitmap fonts? They already have a nicely packed texture atlas with all the characters in it, so it might be a waste of memory/effort to generate one on the fly.
    * The exception to this would be for multi-page fonts - generating a combined atlas would allow those to be used with much less texture swapping. But at the same time, I don't think they're that widely used - LibGDX doesn't support them, I don't think.
